### PR TITLE
Allow for hanging indent when documenting args in Google style

### PR DESCRIPTION
**Pull request by rmorshea**
Closes: #449 

Please make sure to check for the following items:
- [x] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.

### CODE REVIEW

File: docs/release_notes.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ Current Development Version
 
 New Features
 
+* Allow for hanging indent when documenting args in Google style. (#554)
```
Comment (SyntaxColoring) [inline docs/release_notes.rst:13]: Looks like this should say `(#449)`, not `(#554)`.

File: src/pydocstyle/checker.py [hunk 2]
```diff
@@ -122,6 +123,8 @@ class ConventionChecker:
         r"\s*"
         # Followed by a colon
         r":"
+        # Might have a new line
+        r"\n?",
```
Comment (sigmavirus24) [inline src/pydocstyle/checker.py:127]: Did you test this? There shouldn't be a comma at the end of this line, it's not a tuple, it's a regular expression relying on string concatenation over multiple lines with comments to make it more readable.

Comment (rmorshea) [reply src/pydocstyle/checker.py:127]: Yup, I was making some changes through the browser editor and didn't catch this. Will fix shortly.

**Comment by rmorshea** [comment]
Tests should pass now.

**Comment by rmorshea** [comment]
@SyntaxColoring @sigmavirus24 any change of a re-review on this?

File: src/tests/test_cases/sections.py [hunk 3]
```diff
@@ -367,9 +367,6 @@ def test_missing_docstring(a, b):  # noqa: D213, D407
         """
 
     @staticmethod
-    @expect("D417: Missing argument descriptions in the docstring "
-            "(argument(s) skip, verbose are missing descriptions in "
-            "'test_missing_docstring_another' docstring)", arg_count=2)
     def test_missing_docstring_another(skip, verbose):  # noqa: D213, D407
```
Comment (samj1912) [inline src/tests/test_cases/sections.py:370]:
```suggestion
    def test_hanging_indent(skip, verbose):  # noqa: D213, D407
```

**Comment by rmorshea** [comment]
Looks like the `pip install` just failed. Maybe re-running CI again will just pass?

**Review by samj1912 — APPROVED** [review]

**Comment by rmorshea** [comment]
Thanks @samj1912

**Pull request merged**